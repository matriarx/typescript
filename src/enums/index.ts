export enum Environment {
  DEVELOPMENT = 'development',
  TEST = 'test',
  QA = 'qa',
  ALPHA = 'alpha',
  BETA = 'beta',
  UAT = 'uat',
  STAGE = 'stage',
  PRODUCTION = 'production',
  DEMO = 'demo',
}

export enum Path {
  INDEX = '/',
  AUTH = '/auth',
}

export default {
  Environment,
  Path,
}
