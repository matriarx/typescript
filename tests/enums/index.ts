import {Environment, Path} from 'enums'

describe('Enums', () => {
  test('export default', async () => {
    const enums = (await import('enums')).default

    expect(enums).toMatchSnapshot()
  })

  test('enum Environment', async () => {
    expect('development' as Environment).toStrictEqual(Environment.DEVELOPMENT)
    expect('test' as Environment).toStrictEqual(Environment.TEST)
    expect('qa' as Environment).toStrictEqual(Environment.QA)
    expect('alpha' as Environment).toStrictEqual(Environment.ALPHA)
    expect('beta' as Environment).toStrictEqual(Environment.BETA)
    expect('uat' as Environment).toStrictEqual(Environment.UAT)
    expect('stage' as Environment).toStrictEqual(Environment.STAGE)
    expect('production' as Environment).toStrictEqual(Environment.PRODUCTION)
    expect('demo' as Environment).toStrictEqual(Environment.DEMO)
    expect(Environment).toMatchSnapshot()
  })

  test('enum Path', async () => {
    expect('/' as Path).toStrictEqual(Path.INDEX)
    expect('/auth' as Path).toStrictEqual(Path.AUTH)
    expect(Path).toMatchSnapshot()
  })
})
