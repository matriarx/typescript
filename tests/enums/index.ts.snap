// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Enums enum Environment 1`] = `
Object {
  "ALPHA": "alpha",
  "BETA": "beta",
  "DEMO": "demo",
  "DEVELOPMENT": "development",
  "PRODUCTION": "production",
  "QA": "qa",
  "STAGE": "stage",
  "TEST": "test",
  "UAT": "uat",
}
`;

exports[`Enums enum Path 1`] = `
Object {
  "AUTH": "/auth",
  "INDEX": "/",
}
`;

exports[`Enums export default 1`] = `
Object {
  "Environment": Object {
    "ALPHA": "alpha",
    "BETA": "beta",
    "DEMO": "demo",
    "DEVELOPMENT": "development",
    "PRODUCTION": "production",
    "QA": "qa",
    "STAGE": "stage",
    "TEST": "test",
    "UAT": "uat",
  },
  "Path": Object {
    "AUTH": "/auth",
    "INDEX": "/",
  },
}
`;
